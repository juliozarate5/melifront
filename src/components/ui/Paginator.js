import React from 'react';
import PropTypes from 'prop-types';

export default function Paginator(
    {pagination, arrayTotal, handlePrevious, handlePage, handleNext}
    ) {
    return (
        <nav aria-label="paginaciÃ³n" className="my-3">
        <ul className="pagination justify-content-center">
            <li className={`page-item ${pagination.pageNumber==0?"disabled":""}`}>
                <a 
                    onClick={handlePrevious} 
                    className="page-link"
                >
                    Anterior
                </a>
            </li>
            {
               arrayTotal.map((e, i) => (
                <li 
                    key={i}
                    className={`page-item ${pagination.pageNumber==i?"disabled":""}`}
                    >
                 <a 
                    onClick={() => handlePage(i)} 
                    className="page-link" 
                 >
                  {i+1}
                 </a>
                </li>
                )
               )
            }                       
    
            <li className={`page-item ${pagination.pageNumber==pagination.totalPages-1?"disabled":""}`}>
                <a 
                    onClick={handleNext} 
                    className="page-link" 
                    
                >
                    Siguiente
                </a>
            </li>
        </ul>
        </nav>
    )
}


Paginator.propTypes = {
    pagination: PropTypes.shape({
        pageNumber: PropTypes.number,
        totalPages: PropTypes.number,
    }).isRequired, 
    arrayTotal: PropTypes.array.isRequired, 
    handlePrevious: PropTypes.func.isRequired, 
    handlePage: PropTypes.func.isRequired, 
    handleNext: PropTypes.func.isRequired
}
