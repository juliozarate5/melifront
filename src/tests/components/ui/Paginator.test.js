import React from "react";
import { shallow } from "enzyme";
import Paginator from "../../../components/ui/Paginator";

describe('Paginator tests', () => {

    const pagination = {}; 
    const arrayTotal = [];
    const handlePrevious = jest.fn(); 
    const handlePage = jest.fn();
    const handleNext = jest.fn();

    const wrapper = shallow(<Paginator 
        pagination={pagination}
        arrayTotal={arrayTotal}
        handlePrevious={handlePrevious}
        handlePage={handlePage}
        handleNext={handleNext}
    />);

    test('showing component successful', () => {
        expect(wrapper).toMatchSnapshot()
    });

    test('must change paginationNumber', () => {

        const a = wrapper.find('a.page-link').first();

        a.simulate('click');        
    })
})