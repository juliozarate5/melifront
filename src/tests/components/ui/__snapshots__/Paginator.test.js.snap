// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator tests showing component successful 1`] = `
<nav
  aria-label="paginaciÃ³n"
  className="my-3"
>
  <ul
    className="pagination justify-content-center"
  >
    <li
      className="page-item "
    >
      <a
        className="page-link"
        onClick={[MockFunction]}
      >
        Anterior
      </a>
    </li>
    <li
      className="page-item "
    >
      <a
        className="page-link"
        onClick={[MockFunction]}
      >
        Siguiente
      </a>
    </li>
  </ul>
</nav>
`;
